riptide-core/src/main/java/org/zalando/riptide/RequestArguments.java

```diff
@@ -95,8 +97,8 @@ default RequestArguments withRequestUri() {
         {
             final UriComponentsBuilder components = UriComponentsBuilder.newInstance();
             getQueryParams().entries().forEach(entry ->
-                    components.queryParam(entry.getKey(), entry.getValue()));
-            queryParams = components.build().encode().getQueryParams();
+                    components.queryParam(entry.getKey(), UriUtils.encode(entry.getValue(), Charset.defaultCharset())));
```

Comment:
That won't compile in Spring 4, I believe since the overloaded version with `Charset` didn't exist back then.

This should work:

```java
getQueryParams().entries().forEach(throwingConsumer(entry ->
        components.queryParam(entry.getKey(), encode(entry.getValue(), "UTF-8"))));
```